packages/firestore/src/core/target.ts

```diff
@@ -536,6 +536,12 @@ export function targetGetSegmentCount(target: Target): number {
 
 export abstract class Filter {
   abstract matches(doc: Document): boolean;
+
+  abstract getFlattenedFilters(): FieldFilter[];
```

Comment:
Yes! This would improved the readability/navigation of the code. However, I want to do this refactoring in another PR with no other changes. I'm adding a TODO for this refactoring.

Reply:
👍

Reply:
Assigned a task to you for this.